packages/starknet-snap/src/signTransaction.ts

```diff
@@ -0,0 +1,27 @@
+import { Signature, Signer, stark } from 'starknet';
+import { ApiParams, SignTransactionParams } from './types/snapApi';
+import { getKeysFromAddress } from './utils/starknetUtils';
+import { getNetworkFromChainId } from './utils/snapUtils';
+
+export async function signTransaction(params: ApiParams): Promise<Signature> {
+  const { state, keyDeriver, requestParams } = params;
+  const requestParamsObj = requestParams as SignTransactionParams;
+  const userAddress = requestParamsObj.userAddress;
+  const network = getNetworkFromChainId(state, requestParamsObj.chainId);
+  const { privateKey } = await getKeysFromAddress(keyDeriver, network, state, userAddress);
+  const signer = new Signer(privateKey);
```

Comment:
dialog with sign information should popup to ask user to consensus

Reply:
+1!

Reply:
TBC